File: plantio.domain/Users/UserErrors.cs
using plantio.Domain.Generic;

namespace plantio.Domain.Users {
    public static class UserErrors {
        public static readonly DomainError NotFound =
            new DomainError(ErrorCode.UserNotFound, "User not found", "Create a user and try again");
        public static readonly DomainError AlreadyExist =
            new DomainError(ErrorCode.UserAlreadyExist, "User already exists", "Use the existing user instead");
        public static readonly DomainError InvalidCredentials =
            new DomainError(ErrorCode.UserInvalidCredentials, "User's name or password are invalid", "Retry");
    }
}

File: plantio/src/main/java/com/deltegui/plantio/users/application/UserErrors.java
package com.deltegui.plantio.users.application;

import com.deltegui.plantio.common.DomainError;

public enum UserErrors implements DomainError {
    NotFound(100, "User not found"),
    AlreadyExsists(101, "User already exists"),
    InvalidPassword(102, "Invalid password"),
    BagOverflow(103, "Too many items in bag");
    private final int code;
    private final String message;

    UserErrors(int code, String message) {
        this.code = code;
        this.message = message;
    }

    @Override
    public int getCode() {
        return code;
    }

    @Override
    public String getMessage() {
        return message;
    }
}
